draft-edm-protocol-greasing.md

```diff
@@ -99,66 +99,56 @@ designing for and/or implementing version negotiation and migration.
-restrictions.
-
-It is beneficial to have a large set of reserved numbers for the purpose of
-greasing. However, protocol designers that wish to do so may encounter
-difficulties in expressing the large range in their protocol documents and/or in
-an IANA registry. One approach to this problem has been to define the set
-algorithmically in the protocol definition and request that IANA reserve only a
-single entry in the respective table that covers the entire range; see for
-example
+Many protocols register values, codepoints, or numbers in a limited space. A
+common approach is to reserve a subset of the space for greasing (see
+{{GREASE}}, {{Section 18.1 of QUIC}}, or {{Section 7.2.8 of RFC9114}}). Values
+reserved for the purpose of greasing are herein referred to as grease values.
+Implementations that receive grease values are required to ignore them. More
+background to this approach is given in {{Section 3.3 of ?VIABILITY=RFC9170}}.
+This section provides concrete suggestions for its usage.
+
+It is assumed that endpoints should implement robust and broad extension
+handling. A receiver or a parser implementation should not treat grease values
+as individually special. Instead of identifying each grease value explicitly,
+it is better to have a "catch all" mechanism that can handle receipt of unknown
+extensions, whether grease values or not, gracefully or without error.
+
+It is recommended that senders pick an unpredictable grease value to include in
+relevant protocol elements. This ensures that receiver greasing requirements are
+exercised. Using predictable grease values risks ossification. To increase the
+variety of grease values, it is advised to reserve a large range. However, the
+specific size and distribution of the grease range needs to accommodate the
+protocol constraints. For instance, protocols that use 8-bit fields may find it
+too costly to dedicate many grease values, while 32-bit or 64-bit fields are
+likely to have no limitations.
+
+It is recommended that senders use grease values at unpredictable times or
+sequence points during protocol interactions. This avoids receivers
+unintentionally ossifying on the occurrence of greasing in the temporal or
+spatial domain.
+
+It is recommended that large grease value sets are allocated in protocol
+documents by defining a unique algorithm, this increases the chance that
+receiver greasing requirements are exercised. However, the choice of algorithm
+needs to consider the spread of values and the size of contiguous blocks between
+grease values. It is common for protocol extension designers to want to reserve
+a contiguous block of code points in order to aid iteration and experimentation.
+Small contiguous blocks increase the chance that such reservations might
+unintentionally use grease values, which could lead to interoperability
+failures.
+
+Fields that use grease values should request that IANA requests allocate only a
```

Comment:
```suggestion
Fields that use greased values should request that IANA requests allocate only a
```
Is this right?

Reply:
Or greasing?

Reply:
I was attempting keep keep consistency with

> Values
reserved for the purpose of greasing are herein referred to as grease values.

 but really, I don't think "fields that x" can request anything from IANA because they are inanimate objects :joy:. Will take another stab.